modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java

```diff
@@ -538,8 +559,12 @@ private int _authenticate(
 			}
 		}
 
-		return _authenticateRequired(
-			companyId, userId, emailAddress, screenName, true, FAILURE);
+		if (!userImportedByLDAP) {
+			return _authenticateRequired(
+				companyId, userId, emailAddress, screenName, true, FAILURE);
+		}
+
+		return FAILURE;
```

Comment:
Can you help me what your changes do here? It clearly changes the result fundamentally but I don't really understand why.

Reply:
I noticed that there is a odd behavior on this block since the ldap server authentication fails and even then the user is logged successfully.